src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesReconciliationService.kt

```diff
@@ -197,24 +202,34 @@ class CaseNotesReconciliationService(
         .groupBy { it.text.take(maxLengthToCompare) + it.occurrenceDateTime + it.creationDateTime + it.type + it.subType }
         .filter { it.value.size == 2 }
         .forEach { (_, identicalCaseNotePair) ->
-          val id1 = identicalCaseNotePair.first().legacyId
-          val id2 = identicalCaseNotePair.last().legacyId
-          val inDps1 = dpsCaseNotes.values.any { it.legacyId == id1 }
-          val inDps2 = dpsCaseNotes.values.any { it.legacyId == id2 }
-          val inMapping1 = mappings.any { it.nomisCaseNoteId == id1 }
-          val inMapping2 = mappings.any { it.nomisCaseNoteId == id2 }
-          if (inDps1 && !inDps2 && inMapping1 && !inMapping2) {
-            caseNotesNomisApiService.deleteCaseNote(id2)
-            telemetryClient.trackEvent(
-              "casenotes-reports-reconciliation-mismatch-deleted",
-              mapOf("offenderNo" to offenderNo, "nomisId" to id2.toString()),
-            )
-          } else if (!inDps1 && inDps2 && !inMapping1 && inMapping2) {
-            caseNotesNomisApiService.deleteCaseNote(id1)
-            telemetryClient.trackEvent(
-              "casenotes-reports-reconciliation-mismatch-deleted",
-              mapOf("offenderNo" to offenderNo, "nomisId" to id1.toString()),
-            )
+          val nomisCaseNote1 = identicalCaseNotePair.first()
+          val nomisCaseNote2 = identicalCaseNotePair.last()
+          val textToCompare = nomisCaseNote1.text.take(maxLengthToCompare)
+          val exactlyOneExistsInDps = dpsCaseNotes.values.filter {
+            it.text.take(maxLengthToCompare) == textToCompare &&
+              it.occurrenceDateTime == nomisCaseNote1.occurrenceDateTime &&
+              it.creationDateTime == nomisCaseNote1.creationDateTime &&
+              it.type == nomisCaseNote1.type &&
+              it.subType == nomisCaseNote1.subType
+          }
+            .size == 1
```

Comment:
Altered logic to find DPS case note using details as the legacy id is deprecated and cannot be relied on